### DISCUSSION TITLE
Copilot chat no longer working with Visual Studio Code Insiders

### DISCUSSION THREAD

**Discussion by AsaBattle** [community/community#103051] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I've tried uninstalling and reinstalling, but this does not work. Regular copilot works fine. But chat always gives me this:
unhandled status from server: 410 {"error":{"agent-version":"","details":"https://github.blog/changelog/label/copilot/#%F0%9F%8C%90-upcoming-deprecation-of-copilot-chat-api-endpoints","editor-plugin-version":"copilot-chat/0.6.0","editor-version":"vscode/1.81.0-insider","engine":"copilot-chat","message":"copilot-proxy.githubusecontent.com is no longer accepting chat requests. Please update your IDE extension to continue using GitHub Copilot chat."}}

**Comment by hcadioli** [comment, marked as answer]
## Fix

I have fixed it by updating my VSCode to the latest version 1.86
https://code.visualstudio.com/updates/v1_86
After that, I've update my copilot extension and it's now working fine


## Context
Getting the error: 
```
unhandled status from server: 410 {"error":{"agent-version":"","details":"https://github.blog/changelog/label/copilot/#%F0%9F%8C%90-upcoming-deprecation-of-copilot-chat-api-endpoints","editor-plugin-version":"copilot/0.2.2","editor-version":"vscode/1.79.2","engine":"copilot-chat","message":"copilot-proxy.githubusecontent.com is no longer accepting chat requests. Please update your IDE extension to continue using GitHub Copilot chat."}}
```

From [github's](https://github.blog/changelog/label/copilot/#%F0%9F%8C%90-upcoming-deprecation-of-copilot-chat-api-endpoints) official website

As we [announced in September](https://github.blog/changelog/2023-09-25-breaking-change-to-github-copilot-service-endpoints/), we updated the API service endpoints used by Copilot Chat. On February 1, 2024, we will deprecate the Copilot Chat API endpoints currently being routed through https://copilot-proxy.githubusercontent.com. Instead, these requests will go through https://api.githubcopilot.com. Versions 0.8.0 and later of the Copilot Chat extension for VS Code and versions 0.1.1817.27579 and later of the Copilot Chat extension for Visual Studio already route chat traffic through https://api.githubcopilot.com. All versions of the Copilot plugin for JetBrains IDEs already route chat traffic through https://api.githubcopilot.com.

**Reply by davecheney** [reply]
@hcadioli @DaigaroCota newer versions of the Copilot extensions make use of features available in recent versions of VSCode, you should keep on top of the upgrades for both the client extension, and VSCode itself as lagging behind in the latter may limit the versions of the extension available to you.

**Reply by baradhili** [reply]
I am running code-insiders, it is not working as of today!! no response, the only idiciation I have that something is wrong is that I can't open an inline chat. I am running Visual Studio COde - Insiders version 1.87.0-insider - dated 2024-02-20

**Reply by davecheney** [reply]
@baradhili this is a different issue, the issue that @hcadioli had was running an outdated extension. Please create a new discussion post.

**Comment by 0xdcota** [comment]
I am having the same problem here too.

**Reply by 0xdcota** [reply]
Like why don't make breaking updates on WEEKENDS!!!

**Reply by hcadioli** [reply]
I've managed to fix it here:
https://github.com/orgs/community/discussions/103051#discussioncomment-8340729

**Reply by maximilianhagerf** [reply]
Copilot autocomplete is working for me, but when I open the Copilot Chat all I get is a button that says "Start 30-day Free Trial".  I Even switched the Chat extension to pre-release... 

Code info:
Version: 1.86.0
Commit: 05047486b6df5eb8d44b2ecd70ea3bdf775fd937
Date: 2024-01-31T10:29:11.933Z
Electron: 27.2.3
ElectronBuildId: 26495564
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.2.0

**Reply by davecheney** [reply]
@maximilianhagerf this is an unrelated issue. please contact support https://support.github.com/contact.

**Reply by amiroff** [reply]
Having the same issue as @maximilianhagerf after updating VSCode and Copilot extensions to latest stable versions.
Edit: Removing VSCode and installing it again fixed the problem for me.

**Reply by timenauters** [reply]
Same that @maximilianhagerf, until reinstall vs and relogged here

**Comment by tmccormack165** [comment]
To update VS Code:
1. Go to [download page](https://code.visualstudio.com/Download#)
2. Select the version
3. Go through the installer helper on your machine, if on linux use these commands:
```
cd ~/Downloads && sudo dpkg -i code*
```
4. Open VS Code and Go to Extensions, for Github Copilot you will see a blue button that says _Reload Required_, press that button and go to chat, and it should work, good luck.

**Reply by davecheney** [reply]
Thank you. Older versions of VSCode will not show newer versions of extensions if those extensions require features not available in the older VSCode. I recommend updating VSCode regularly -- they have a monthly stable release cadence.

**Comment by davecheney** [comment]
Duplicate of https://github.com/orgs/community/discussions/101438

**Comment by edson-arcaea** [comment]
I had this problem and tried the solution suggested above and it didn't work. Ultimately, I completely uninstalled VS Code and re-installed it again then pressed reload on copilot chat and it works now.
